File: Ejercicio personajes (Fabrica abstracta)/Fabrica abstracta V2/src/Fabricas/FabricaHumanos.java
package Fabricas;

import Personajes.ArmaduraHumano;
import Personajes.ArmaHumano;
import Personajes.CuerpoHumano;
import Personajes.MonturaHumano;
import Interfaces.IArmadura;
import Interfaces.IArma;
import Interfaces.ICuerpo;
import Interfaces.IMontura;

public class FabricaHumanos extends FabricaAbstracta {

    @Override
    public IArma crearArma() {
        return new ArmaHumano();
    }

    @Override
    public IArmadura crearArmadura() {
        return new ArmaduraHumano();
    }

    @Override
    public IMontura crearMontura() {
        return new MonturaHumano();
    }

    @Override
    public ICuerpo crearCuerpo() {
        return new CuerpoHumano();
    }
}

File: Ejercicio personajes (Fabrica abstracta)/CombatChoose_V2/fabricas/fabrica_humanos.py
from fabricas.fabrica_abstracta import FabricaAbstracta
from interfaces.arma import IArma
from interfaces.armadura import IArmadura
from interfaces.cuerpo import ICuerpo
from interfaces.montura import IMontura

class Espada(IArma):
    def get_descripcion(self):
        return "Espada de acero"

class ArmaduraPlacas(IArmadura):
    def get_descripcion(self):
        return "Armadura de placas"

class CuerpoHumano(ICuerpo):
    def get_descripcion(self):
        return "Físico balanceado"

class Caballo(IMontura):
    def get_descripcion(self):
        return "Caballo de guerra"

class FabricaHumanos(FabricaAbstracta):
    def crear_arma(self):
        return Espada()
    def crear_armadura(self):
        return ArmaduraPlacas()
    def crear_cuerpo(self):
        return CuerpoHumano()
    def crear_montura(self):
        return Caballo()
